CMakeLists.txt

```diff
@@ -1,22 +1,65 @@
+cmake_minimum_required(VERSION 3.22)
 PROJECT( twbt )
+set( TWBT_VER "6.xx" )
 
-SET( PROJECT_SRCS twbt.cpp )
-SET( TWBT_VER "5.xx" )
+SET(CMAKE_VERBOSE_MAKEFILE ON)
+if(DEFINED dfhack_SOURCE_DIR)
+    set(DFHACK ${dfhack_SOURCE_DIR})
+elseif(EXISTS ../dfhack) # sitting next to each other top level
+    set(DFHACK ../dfhack)
+elseif(EXISTS ../../dfhack) # twbt is inside a dfhack directory, probably dfhack/plugins
+    set(DFHACK ../../dfhack)
+elseif(EXISTS ../../../dfhack) # twbt is inside a directory in dfhack/plugins (probably)
+    set(DFHACK ../../../dfhack)
+else()
+    message("couldn't find a dfhack directory nearby, modify the cmake and put your dfhack path")
+    SET(DFHACK /your/path/dfhack)
+endif()
 
-STRING( REGEX REPLACE "\\." "" DV_VS "${DF_VERSION}" )
+include(${DFHACK}/CMake/Common.cmake)
+message("Version information\n-------------------")
+message("dwarf fortress: ${DF_VERSION}")
+message("dfhack:         ${DFHACK_VERSION}")
+message("twbt:           ${TWBT_VER}")
+string(REPLACE "." "" DF_VERSION ${DF_VERSION})
 
-SET( CMAKE_CXX_FLAGS "-m64 -O3 -DDFHACK_VERSION=\"${DFHACK_VERSION}\" -DDF_${DV_VS} -DTWBT_VER=\"\\\"${TWBT_VER}\\\"\"" )
+# include directories
+include_directories(${DFHACK}/library/include)
+include_directories(${DFHACK}/library/proto)
+include_directories(${DFHACK}/depends/protobuf)
+include_directories(${DFHACK}/depends/lua/include)
```

Comment:
Redundant if df-twbt is inside dfhack/plugins

Reply:
I didn't want to complicate the conditional logic so I went for redundancy